worker/changestream/eventmultiplexer/eventmultiplexer.go

```diff
@@ -181,16 +190,18 @@ func (e *EventMultiplexer) loop() error {
 		}
 		e.subscriptions = nil
 		e.subscriptionsByNS = nil
-
-		close(e.subscriptionCh)
-		close(e.unsubscriptionCh)
 	}()
 
 	for {
 		select {
+		// If the catacomb is dying, then we should exit.
 		case <-e.catacomb.Dying():
 			return e.catacomb.ErrDying()
 
+		// If the underlying stream is dying, then we should also exit.
+		case <-e.stream.Dying():
+			return nil
```

Comment:
This is aping a graceful shut-down, which will prevent third party worker management from starting us back up. I think it should be something like:
```go
case <-e.stream.Dying():
    <-e.catacomb.Dying()
    return e.catacomb.ErrDying()
```